addon/addon/services/animations.ts

```diff
@@ -25,13 +28,21 @@ export type AnimateFunction = (
   motion: Motion
 ) => SpriteAnimation;
 
-function filterToContext(
+export function filterToContext(
   spriteTree: SpriteTree,
   animationContext: AnimationContext,
   spriteModifiers: Set<SpriteModifier>,
-  opts = { includeFreshlyRemoved: false }
+  opts: GetDescendantNodesOptions = { includeFreshlyRemoved: false }
 ): Set<SpriteModifier> {
-  let contextDescendants = spriteTree.descendantsOf(animationContext, opts);
+  let contextDescendants = spriteTree.descendantsOf(animationContext, {
+    ...opts,
+    filter(childNode: SpriteTreeNode) {
+      return !(
+        childNode.isContext &&
+        (childNode.contextModel as AnimationContext).isStable
+      );
+    },
+  });
   let result = new Set(
     [...spriteModifiers].filter((m) => contextDescendants.includes(m))
   );
```

Comment:
Note to self that these changes will likely break interruptions for nested animation contexts in certain ways. We will rework the interruptions story (in willTransition) to work with that situation later.